provider/cluster/inventory.go

```diff
@@ -315,7 +315,9 @@ loop:
 
 func (is *inventoryService) runCheck() <-chan runner.Result {
 	return runner.Do(func() runner.Result {
-		return runner.NewResult(is.client.Inventory())
+		// TODO: Handle shutdown via context.Context.
```

Comment:
can make one based off of `is.lc.ShuttingDown()` or just from a `cancel()` in the mainloop.